components/selector/__init__.py

```diff
@@ -217,7 +218,7 @@ def _calculate_observation_group(self,
         night_events = self.collector.get_night_events(obs.site)
 
         for night_idx in ranker.night_indices:
-            actual_conditions = self.collector.get_actual_conditions_variant(obs.site, night_events.times[night_idx])
+            actual_conditions = Selector.static_Env.get_actual_conditions_variant(obs.site, night_events.times[night_idx])
```

Comment:
...which would make this:
```
actual_conditions = Selector._env.get_actual_conditions_variant(obs.site, night_events.times[night_idx])
```

So, great job. You did it just right and more than I asked, so if you make these changes, we can merge.